Sources/Purchasing/StoreKitAbstractions/PromotionalOffer.swift

```diff
@@ -42,15 +43,30 @@ public final class PromotionalOffer: NSObject {
 
 // MARK: - SignedData
 
-extension PromotionalOffer {
+@objc public extension PromotionalOffer {
 
     /// Contains the details of a promotional offer discount that you want to apply to a payment.
-    internal struct SignedData {
-        var identifier: String
-        var keyIdentifier: String
-        var nonce: UUID
-        var signature: String
-        var timestamp: Int
+    @objc(RCPromotionalOfferSignedData)
+    class SignedData: NSObject {
+        /// The subscription offer identifier.
+        @objc public private(set) var identifier: String
+        /// The key identifier of the subscription key.
+        @objc public private(set) var keyIdentifier: String
+        /// The nonce used in the signature.
+        @objc public private(set) var nonce: UUID
+        /// The cryptographic signature of the offer parameters, generated on RevenueCat's server.
+        @objc public private(set) var signature: String
+        /// The UNIX time, in milliseconds, when the signature was generated.
+        @objc public private(set) var timestamp: Int
+
+        init(identifier: String, keyIdentifier: String, nonce: UUID, signature: String, timestamp: Int) {
+            self.identifier = identifier
+            self.keyIdentifier = keyIdentifier
+            self.nonce = nonce
+            self.signature = signature
+            self.timestamp = timestamp
+            super.init()
```

Comment:
Nit: I think this needs to go first before setting the properties.

Reply:
> Nit: I think this needs to go first before setting the properties.

You only need to call `super.init()` if the subclass's initializer customizes some properties it is inheriting from (and it would go first). In this case, you can omit the call to `super.init()`

https://docs.swift.org/swift-book/LanguageGuide/Initialization.html# specifically search for `you can omit a call to super.init()` and it will bring you to the section that details this. I forget when they added this to swift, but I feel like it was maybe in 4.0.